coordinator/SEs/klee_conc_explorer_minimal.py

```diff
@@ -0,0 +1,250 @@
+
+    def get_config(self):
+        config = ConfigParser.ConfigParser()
+        config.read(self.config)
+        self.bin = config.get("klee conc_explorer_minimal", "bin")
+        self.converter = config.get("klee conc_explorer_minimal","converter")
+        self.seed_dir = config.get("klee conc_explorer_minimal", "klee_seed_dir").replace("@target", self.target)
+        self.target_bc = config.get("moriarty", "target_bc").replace("@target", self.target).split()[0]
+        self.options = config.get("moriarty", "target_bc").replace("@target", self.target).split()[1:]
+        self.klee_err_dir = config.get("klee conc_explorer_minimal", "error_dir").replace("@target", self.target)
+
+        try:
+            self.max_time_per_seed = config.get("klee conc_explorer_minimal", "max_time_per_seed")
+        except Exception:
+            # by default no time limit per seed.
+            self.max_time_per_seed = '0'
+
+        # Time budget per solving attempt
+        try:
+            self.max_solver_time = config.get("klee conc_explorer_minimal", "max_solver_time")
+        except Exception:
+            # by default no time limit per seed.
+            self.max_solver_time = '0'
+
+        try:
+            self.max_mem = config.get("klee conc_explorer_minimal", "max_memory")
+        except Exception:
+            self.max_mem = str(1024*1024*1024) # 1gb
+
+        try:
+            self.fuzzer_cov_file = config.get("auxiliary info", "cov_edge_file").replace("@target", self.target)
+        except Exception:
+            self.fuzzer_cov_file = os.path.join(self.target,".afl_coverage_combination")
+
+        self.bitmodel = config.get("moriarty", "bitmodel")
+        self.input_type = config.get("moriarty", "inputtype")
+        self.sync_dir_base = config.get("moriarty", "sync_dir").replace("@target", self.target)
+
+        
+    def __repr__(self):
+        return "SE Engine: KLEE Concolic Explorer (minimal)"
+
+
+    def get_search_heuristics(self):
+        """return a list of search heuristics"""
+        return []
+
+    def exceed_mem_limit(self):
+        pass
+
+    def alive(self):
+        alive = False
+        multiprocessing.active_children()
+        for pid in [self.jobs[x]['real_pid'] for x in self.jobs]:
+            try:
+                os.kill(pid, 0)
+                print "conc_explorer pid: {0} is alive".format(pid)
+                alive = True
+            except Exception:
+                print "conc_explorer pid: {0} not alive".format(pid)
+
+        return alive
+
+    def run(self, input_id_map_list, cov_file):
+        """
+            -create seed-out-dir
+            For each input,
+                -convert ktest move to seed-out-dir
+            -create sync dir
+            -build cmd
+            -create new process job
+        """
+        #assert(len(input_id_map_list) == 1) # 1 KLEE per seed
+
+        pid = self.get_new_pid()
+        klee_seed_dir = self.seed_dir + "/klee_instance_conc_"+str(pid)
```

Comment:
I see that klee_conc_explorer.py zero-pads but this doesn't. Hopefully this doesn't break such an assumption somewhere else in the code?